servers/services/src/main/java/org/projectnessie/services/impl/NamespaceApiImpl.java

```diff
@@ -218,6 +225,35 @@ public GetNamespacesResponse getNamespaces(MultipleNamespacesParams params)
     }
   }
 
+  @Override
+  public void updateProperties(NamespaceParams params, NamespaceUpdate namespaceUpdate)
+      throws NessieNamespaceNotFoundException, NessieReferenceNotFoundException {
+    try {
+      BranchName branch = branchFromRefName(params.getRefName());
+
+      Namespace namespace = getNamespace(params.getNamespace(), branch);
+      Map<String, String> properties = new HashMap<>(namespace.getProperties());
```

Comment:
Can be simplified with
```
ImmutableNamespace.Builder namespace = Namespace.builder().from(getNamespace(...));
```
and then iterate over propertyRemovals+propertyUpdates and call the corresponding builder methods.

Reply:
there's only a builder method to add properties but not remove them unfortunately

Reply:
Duh - that's right